app/kuma-dp/cmd/context.go

```diff
@@ -19,6 +19,7 @@ type RootContext struct {
 	ComponentManager         component.Manager
 	BootstrapGenerator       envoy.BootstrapConfigFactoryFunc
 	BootstrapDynamicMetadata map[string]string
+	DataplaneTokenGenerator  func() ([]byte, error)
```

Comment:
Can we drop a comment? Somthing like: it is for generating token on the dp side using environment in which DP is deployed so cp can validate. Currently it's not implemented in Kuma, but can be extended to provide alternative auth mechanism.

I'd be quite confused that we have a logic that is not used

Reply:
An interface for what? The `RootContext` is fundamentally a public API, it is one of the main points where things are overriden.

But having a default generator would help here like you suggest in the other comment.